regulations/views/mixins.py

```diff
@@ -6,7 +6,20 @@
 from regulations.generator import api_reader
 
 
-class SidebarContextMixin:
+class CitationContextMixin:
+    def get_context_data(self, **kwargs):
+        context = super(CitationContextMixin, self).get_context_data(**kwargs)
+        citation = []
+        if 'part' in context:
+            citation.append(context["part"])
+        if 'section' in context:
+            citation.append(context["section"])
+        if len(citation) > 0:
+            context['citation'] = "-".join(citation)
+        return context
+
+
+class SidebarContextMixin(CitationContextMixin):
```

Comment:
Not sure this is the right way to handle dependencies in the mixins. Might be better to use a function for DRY and just use it in both mixins.